### PR TITLE
#713 add an option to allow application developer to ignore the untrusted …

### PR DESCRIPTION
**Pull request by royzhang666**
addresses https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/issues/713

### CODE REVIEW

File: src/AspNet.Security.OAuth.Apple/AppleAuthenticationHandler.cs [hunk 1]
```diff
@@ -342,17 +342,6 @@ protected override async Task<HandleRequestResult> HandleRemoteAuthenticateAsync
...
-        {
-            using var user = JsonDocument.Parse(userJson);
-            var userClaims = ExtractClaimsFromUser(user.RootElement);
-
-            foreach (var claim in userClaims)
-            {
-                identity.AddClaim(claim);
-            }
-        }
-
```
Comment (martincostello) [inline src/AspNet.Security.OAuth.Apple/AppleAuthenticationHandler.cs:355]: This breaks things in the case of the form post flow (the default) where the parameters aren't in the query string.

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/4b6591108b5d576005a7115e244b28f34339183c/src/AspNet.Security.OAuth.Apple/AppleAuthenticationHandler.cs#L177-L181

Comment (martincostello) [reply src/AspNet.Security.OAuth.Apple/AppleAuthenticationHandler.cs:355]: I think the proper way to remove this code would be to have the override just call the base method, and the duplicated implementation that supports query and form can just be deleted.

Comment (royzhang666) [reply src/AspNet.Security.OAuth.Apple/AppleAuthenticationHandler.cs:355]: it is still valid, as it ignores the user field from the form post deliberately.

Comment (royzhang666) [reply src/AspNet.Security.OAuth.Apple/AppleAuthenticationHandler.cs:355]: Indeed, remove the boilerplate code and reuse the base would be a good idea. As long as the apple specific design be removed.

**Comment by martincostello** [comment]
Thanks for your PR - we're resolving this through #716 and #718.

**Pull request closed without merging**